lib/debug-server.js

```diff
@@ -85,11 +106,40 @@ DebugServer.prototype.close = function() {
   }
 };
 
-DebugServer.prototype.address = function() {
-  var address = this._httpServer.address();
+DebugServer.prototype.getHttpServerAddressInfo = function() {
+  var addressInfo = this._httpServer.address();
+
+  if(this._shouldAnnounceEndpoint()) {
+    // Resolve local IP, in order for computers in the network to resolve
+    addressInfo.address = localIp();
```

Comment:
Local IP should be used only when the address is `0.0.0.0`. When the user explicitly specifies a public IP, then this IP should be honoured in the broadcasted event.